sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -703,10 +884,14 @@ public void validate(PipelineOptions options) {
             getFlattenResults() != null, "flattenResults should not be null if query is set");
         checkArgument(getUseLegacySql() != null, "useLegacySql should not be null if query is set");
       }
-      checkArgument(getParseFn() != null, "A parseFn is required");
 
       Pipeline p = input.getPipeline();
       final Coder<T> coder = inferCoder(p.getCoderRegistry());
+      if (getMethod() == Method.BQ_PARALLEL_READ) {
+        SerializableUtils.serializeToByteArray(createSource("", coder));
```

Comment:
Why the serialize call?

Reply:
I'm not sure. @yirutang can you comment here?

Reply:
I removed this.